primitives/consensus/common/src/block_import.rs

```diff
@@ -142,13 +142,17 @@ pub struct BlockImportParams<Block: BlockT, Transaction> {
 	/// Is this block finalized already?
 	/// `true` implies instant finality.
 	pub finalized: bool,
+	/// Intermediate values that are interpreted by block importers. Each block importer,
+	/// when handled a value, removes it from the intermediate list. The final block importer
+	/// rejects block import if there are still intermediate values remain unhandled.
+	pub intermediates: HashMap<Vec<u8>, Vec<u8>>,
 	/// Auxiliary consensus data produced by the block.
 	/// Contains a list of key-value pairs. If values are `None`, the keys
 	/// will be deleted.
 	pub auxiliary: Vec<(Vec<u8>, Option<Vec<u8>>)>,
 	/// Fork choice strategy of this import. This should only be set by a
 	/// synchronous import, otherwise it may race against other imports.
-	pub fork_choice: ForkChoiceStrategy,
+	pub fork_choice: Option<ForkChoiceStrategy>,
```

Comment:
Please add docs describing the behavior of the final importer if `None` is supplied.